src/main/scala/tech/cryptonomic/conseil/generic/chain/DataTypes.scala

```diff
@@ -277,14 +301,58 @@ object DataTypes {
       predicate.map(_.into[Predicate].withFieldConst(_.field, field).transform)
   }
 
+  /** Class representing snapshot */
+  case class Snapshot(field: String, value: Any) {
+
+    /** Transforms snapshot to predicate */
+    def toPredicate: Predicate =
+      Predicate(
+        field = field,
+        operation = OperationType.after,
```

Comment:
why is that the correct operation type?

Reply:
`after` is the same as `gt`, if you noticed it has also `inverse` set to `true`.
We want to tae the latest snapshot field by rank, and combining this predicate wit ordering we get that.